### PR TITLE
chore(jest): Manage User Registration Use-Cases [AM-179]

### PR DESCRIPTION
**Pull request by leleueri**

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/logback.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
         <appender-ref ref="STDOUT" />
     </appender>
 
-    <logger name="io.gravitee" level="INFO" />
+    <logger name="io.gravitee" level="DEBUG" />
```
Comment (ashraf706) [inline gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/logback.xml:52]: Is it an intentional log level change from INFO to DEBUG?

File: gravitee-am-test/api/config/dev.setup.js [hunk 2]
```diff
@@ -23,3 +23,7 @@ process.env.AM_DEF_ENV_ID = "DEFAULT";
 process.env.AM_ADMIN_USERNAME = "admin";
 process.env.AM_ADMIN_PASSWORD = "adminadmin";
 process.env.FAKE_SMTP = "http://localhost:5080";
+// here is some Settings that allow Jest Tests execution on Releational DB
```
Comment (ashraf706) [inline gravitee-am-test/api/config/dev.setup.js:26]: spelling: "Relational"

File: gravitee-am-test/specs/gateway/user-registration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+let accessToken;
+let defaultIdp;
+let customIdp;
+let application;
+let clientId;
+let openIdConfiguration;
+
+jest.setTimeout(200000)
+
+beforeAll(async () => {
```
Comment (ashraf706) [inline gravitee-am-test/specs/gateway/user-registration.spec.ts:45]: the file name should contain "jest.spec.ts" to align with rest of the test names. This comment is not relates to this line. Rather it is relates to the test file name :) . You missed the "jest" word

Suggestion: Also noticed the file is not formatted. Would be helpful for future developer if the file is formatted.

File: gravitee-am-test/specs/gateway/user-registration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+        it("When a user register, he should have userId into the session and be redirected to specific URL", async () => {
+            const user = {
+                "firstName": faker.name.firstName(),
+                "lastName": faker.name.lastName(),
+                "username": faker.name.firstName(),
+                "email": faker.internet.email(),
+                "password": "P@ssw0rd!"
+            }
+            const postResponse = await register(user, `https://acustom/web/site?client_id=${clientId}`, true);
```
Comment (ashraf706) [inline gravitee-am-test/specs/gateway/user-registration.spec.ts:249]: unused variable postResponse

File: gravitee-am-test/specs/gateway/user-registration.spec.ts [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+    const createdDomain = await createDomain(accessToken, "user-registration", faker.company.catchPhraseDescriptor());
+    expect(createdDomain).toBeDefined();
+    expect(createdDomain.id).toBeDefined();
+    domain = createdDomain;
+
+    const domainStarted = await startDomain(domain.id, accessToken);
+    expect(domainStarted).toBeDefined();
+    expect(domainStarted.id).toEqual(domain.id);
+    domain = domainStarted;
```
Comment (ashraf706) [inline gravitee-am-test/specs/gateway/user-registration.spec.ts:58]: could be avoid reassigning domainStarted to domain.

**Review by ashraf706 — APPROVED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/logback.xml (hunk 1 above)
Comment (leleueri) [reply gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/logback.xml:52]: no

**Pull request merged**